core/src/main/scala/io/projectglow/vcf/VCFFileFormat.scala

```diff
@@ -273,6 +273,23 @@ object VCFFileFormat {
       false
     }
   }
+
+  def requireWritableAsVCF(schema: StructType): Unit = {
+    val baseRequiredFields =
+      Seq(VariantSchemas.contigNameField, VariantSchemas.startField, VariantSchemas.refAlleleField)
```

Comment:
I believe `endField` is also required as well. To check this more thoroughly, we could have tests that remove each of the columns in a DataFrame and attempt conversion to a `VariantContext`. If conversion fails, we should verify that the check for writable-ness also fails, and vice versa.